src/Microsoft.AspNetCore.Session/DistributedSession.cs

```diff
@@ -210,6 +261,12 @@ private void Deserialize(Stream content)
                 int dataLength = DeserializeNumFrom4Bytes(content);
                 _store[key] = ReadBytes(content, dataLength);
             }
+
+            if (_logger.IsEnabled(LogLevel.Debug))
+            {
+                _sessionId = new Guid(_sessionIdBytes).ToString();
```

Comment:
We are always writing the id to the cache but why do we have this restriction for reading, i.e only if log is enabled?